### ISSUE TITLE
Validate docker 1.8 release candidates

### ISSUE THREAD

**Issue by vishh** [kubernetes/kubernetes#12214]
Docker has started pushing v1.8 release candidates. We should begin validating the release.

**Comment by vishh** [comment]
@yujuhong: Are you planning on working on this? If not, I can pick it up.

**Comment by yujuhong** [comment]
I'll take a shot at this. @vishh, are either you or @dchen1107 planning to write down some basic guidelines for this? It doesn't have to be a polished, comprehensive doc, but I'd like to have something to start with. Thanks! :-)

**Comment by vishh** [comment]
@yujuhong: I would start with setting up a cluster with v1.8 rc3 and running all our stable e2e tests against that cluster. After that, I'd recommend restarting the daemon to ensure that everything is stable. 
There is a node restart e2e test, which is also critical. 
@dchen1107: Are there any other things to test? Can you share your scripts to deploy a new version of docker?

**Comment by yujuhong** [comment]
FYI, here is the [patch](https://github.com/yujuhong/kubernetes/commit/04ce1a94dd6131d7b444f9c7cac6882239187c87) I used to upgrade docker on the GCE nodes.
The docker v1.8.0rc3 change log is [here](https://github.com/docker/docker/blob/v1.8.0-rc3/CHANGELOG.md).

I am still running e2e, but have already seen some failures. Will investigate further once the tests are completed.

**Comment by dchen1107** [comment]
@yujuhong We don't need scripts now. Several fixes were merged to fix salt already. Now you can simply follow what I did with #12499 to start a cluster with new docker version.

I will start a doc for dev on how to validate a docker version.

**Comment by yujuhong** [comment]
> Now you can simply follow what I did with #12499 to start a cluster with new docker version.

I followed the instructions from the comments, but https://get.docker.com/ubuntu/dists/docker/main/binary-amd64/Packages does not provide a package for 1.8 yet.

**Comment by vishh** [comment]
https://github.com/docker/docker/releases/tag/v1.8.0-rc3

**Comment by yujuhong** [comment]
> https://github.com/docker/docker/releases/tag/v1.8.0-rc3

Yep, I saw that. I was just wondering if there is any specific reason to download from https://get.docker.com/ubuntu/dists/docker/main/binary-amd64/Packages  :-)

**Comment by vishh** [comment]
These are release candidates. The final release is not done yet. The intent
is to report any bugs before they make the final release.

**Comment by yujuhong** [comment]
Update: Still running stable tests. Non-stable tests are pretty much meaningless since they fail at head.

Discovered issues so far: #12506

**Comment by yujuhong** [comment]
Update: all but two stable tests passed.
The two failed tests are #12506 and monitoring.go: verify monitoring pods and all cluster nodes are available on influxdb using heapster

```
• Failure [301.456 seconds]
Monitoring
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/monitoring.go:49
  should verify monitoring pods and all cluster nodes are available on influxdb using heapster. [It]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/monitoring.go:48

  Aug 10 17:51:41.548: monitoring using heapster and influxdb test failed

  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/monitoring.go:233
------------------------------
```

Both expected pods, `monitoring-heapster-v7-3o7cc` and `monitoring-influx-grafana-v1-w0sf6`, were running fine through manual check. The test couldn't find them from influxdb. Note that I ran the test twice and both time failed. @vishh, do you have any idea why this test might have failed?

**Comment by vishh** [comment]
Were you able to retrieve any heapster logs?

**Comment by yujuhong** [comment]
> Were you able to retrieve any heapster logs?

https://gist.github.com/yujuhong/f233d8c9b6dd18d1608b

**Comment by yujuhong** [comment]
@vishh, I think it might just be a fluke. I created a new cluster and it passed the same test :\
I am rerunning all stable tests to confirm that the old cluster wasn't in the bad state due to other tests.

**Comment by vishh** [comment]
@yujuhong: From heapster logs it looks like there were many issues - API server watches failing, InfluxDB has no memory, etc. FWIW it might not be worth exploring.

**Comment by yujuhong** [comment]
FWIW, most tests passed in that run :-|

The new run passed all stable tests except for #12506. I will try killing/restarting daemons to see if the cluster would break.

**Comment by vishh** [comment]
https://github.com/docker/docker/releases/tag/v1.8.0

**Comment by bgrant0607** [comment]
1.8 was announced on the blog today:
https://blog.docker.com/2015/08/docker-1-8-content-trust-toolbox-registry-orchestration/

**Comment by vishh** [comment]
https://github.com/docker/docker/pull/15544

**Comment by vishh** [comment]
@yujuhong: 1.8.2-rc1 is out. Do we have some automated way to validate docker now?
https://github.com/docker/docker/releases/tag/v1.8.2-rc1
Linux 64bit binary: https://test.docker.com/builds/Linux/x86_64/docker-1.8.2-rc1

**Comment by yujuhong** [comment]
@vishh, I don't believe we do, other than creating a cluster, deploying docker, and running e2e tests.

I will get to this soon. Perhaps it's time to try creating/keeping multiple e2e clusters?...

**Comment by yujuhong** [comment]
Update:
 - I uploaded a PR to upgrade docker to 1.8.1 (for debian/wheezy) in #13752 
 - I filed #13749 so that we can discuss about future upgrades now that docker release dynamic binaries in deb

**Comment by yujuhong** [comment]
Update: we've upgraded docker to 1.8.2 for testing through #13752.

I also started a cluster of 4 nodes (n1-standard-2) and let it run over the weekend. Besides the add-on pods, each node had 10 extra pods running on it. In the past 12 hours (image below), docker daemon's memory usage went up slightly, but nothing dramatic. There is no obvious sign of memory leakage. I will monitor the soak cluster to confirm that.

![image](https://cloud.githubusercontent.com/assets/10524058/9857743/4cbf4406-5ad1-11e5-97f1-63b783862c85.png)

On some node, docker's memory usage dropped dramatically at some point (~1000MB -> ~600MB ->~150MB), which was probably caused by GC (docker wasn't restarted in the past two days). However, the memory was still charged to the root container and not released. This is nothing new and just re-enforces that we don't have a good handle of the memory usage yet.

/cc @dchen1107
